tests/apps/configurationapp/app.go

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/dapr/components-contrib/configuration"
+	"github.com/dapr/components-contrib/tests/utils/configupdater"
+	cu_redis "github.com/dapr/components-contrib/tests/utils/configupdater/redis"
+	"github.com/dapr/dapr/tests/apps/utils"
+	"github.com/dapr/kit/logger"
+	"github.com/gorilla/mux"
+)
+
+const (
+	daprHost        = "localhost"
+	daprHTTPPort    = "3500"
+	configStoreName = "configstore"
+	redisHost       = "dapr-redis-master.dapr-tests.svc.cluster.local:6379"
+)
+
+var (
+	appPort         = 3000
+	lock            sync.Mutex
+	receivedUpdates map[string][]string
+	updater         configupdater.Updater
+)
+
+type testCommandRequest struct {
+	Message string `json:"message,omitempty"`
+}
+
+type appResponse struct {
+	Message   string `json:"message,omitempty"`
+	StartTime int    `json:"start_time,omitempty"`
+	EndTime   int    `json:"end_time,omitempty"`
+}
+
+type receivedMessagesResponse struct {
+	ReceivedUpdates []string `json:"received-messages"`
+}
+
+func init() {
+	p := os.Getenv("PORT")
+	if p != "" && p != "0" {
+		appPort, _ = strconv.Atoi(p)
+	}
+	receivedUpdates = make(map[string][]string)
+}
+
+// indexHandler is the handler for root path
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	log.Println("indexHandler is called")
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(appResponse{Message: "OK"})
+}
+
+// sendResponse returns response with status code and message
+func sendResponse(w http.ResponseWriter, statusCode int, message string) {
+	w.WriteHeader(statusCode)
+	json.NewEncoder(w).Encode(appResponse{Message: message})
+}
+
+// getKeyValues is the handler for getting key-values from config store
+func getKeyValues(w http.ResponseWriter, r *http.Request) {
+	var keys []string
+	err := json.NewDecoder(r.Body).Decode(&keys)
+	if err != nil {
+		sendResponse(w, http.StatusBadRequest, err.Error())
+		return
+	}
+	url := "http://" + daprHost + ":" + daprHTTPPort + "/v1.0-alpha1/configuration/" + configStoreName + buildQueryParams(keys)
+	resp, err := http.Get(url)
```

Comment:
Please don't use the default HTTP client, which has no timeout and could make the test hang. I belive we have some test utils that perform HTTP requests; otherwise, please just allocate a new, global http.Client